staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -202,21 +198,15 @@ func (q *delayingType) waitingLoop() {
 			delete(waitingEntryByData, entry.data)
 		}
 
-		// Set up a wait for the first item's readyAt (if one exists)
-		nextReadyAt := never
-		if waitingForQueue.Len() > 0 {
-			entry := waitingForQueue.Peek().(*waitFor)
-			nextReadyAt = q.clock.After(entry.readyAt.Sub(now))
-		}
```

Comment:
this might be simpler to follow:
```go
nextReadyAt := never
if waitingForQueue.Len() > 0 {
  if nextReadyAtTimer != nil {
    nextReadyAtTimer.Stop()
  }
  entry := waitingForQueue.Peek().(*waitFor)
  nextReadyAtTimer = time.NewTimer(entry.readyAt.Sub(now))
  nextReadyAt = nextReadyAtTimer.C
}
```

Reply:
👍